src/interpreter.rs

```diff
@@ -18,6 +20,14 @@ impl Display for Value {
             Value::Str(s) => write!(f, "{}", s),
             Value::Num(n) => write!(f, "{}", n),
             Value::Arr() => write!(f, "[]"),
+            Value::Pipeline(out) => write!(
+                f,
+                "{}",
+                match String::from_utf8(out.clone().stdout) {
```

Comment:
This looks like you're reading the entirety of the buffer to a string first to make it work; however, this would require all of stdout being written to app memory first.  Instead, we should wirte our own `display` function

```rust
pub fn display(val: &Value) {
    match val {
        Value::Pipeline(out) => {/* write stdout to stdout */},
        _ => print!("{}", val),
    }
}
```

then call `display` in `main` in place of `print!()`.

Reply:
I see the predicament now--everything is already set up for Display, but Display contradicts the pipe to stdout scenario.  We can perhaps fix in subsequent PR.  I think it will require separating into two functions, ex:
* ```rust
    to_out(val: &Value)
    ```
* ```rust
    std::string::ToString::to_str(val: &Value) -> String
    ```

Reply:
I'll think on this but we should discuss it a bit.